### DISCUSSION TITLE
Setting webhooks in development in AWS Cloud 9

### DISCUSSION THREAD

**Discussion by notjustcodehq** [pay-rails/pay#623] [General] [answered]
I cannot get the Stripe checkout webhooks to work in my development environment which is in AWS Cloud 9. 

I have installed stripe CLI and listen with

`stripe listen --forward-to afcca2cf77754c02b5bab29733b0c1ff.vfs.cloud9.us-east-2.amazonaws.com/pay/webhooks/stripe`

and this seems to work because the Stripe dashboard says the local listener is active.
![webhooks_stripe](https://user-images.githubusercontent.com/35150537/166661239-99899d36-fb3d-46b1-b53e-5e50cba7cf92.jpg)
I also set the :host in the config/development.rb as
`config.action_mailer.default_url_options = { :host => "afcca2cf77754c02b5bab29733b0c1ff.vfs.cloud9.us-east-2.amazonaws.com" }`

STRIPE_PUBLIC_KEY, STRIPE_SECRET_KEY and STRIPE_SIGNING_SECRET are stored as ENV variables using the Figaro gem so in the application.yml file I have:
![application_yml](https://user-images.githubusercontent.com/35150537/166662821-ca7b0f56-9846-4e82-a5fb-df6651af870a.jpg)

The problem is requests return with code 301 and nothing is saved in the database (except the customer).
![webhooks](https://user-images.githubusercontent.com/35150537/166661966-1e937c0a-5c29-4c8f-9ae2-67dd035812a7.jpg)

Any idea what I am doing wrong? Notice that everything works fine in production.

**Comment by excid3** [comment, marked as answer]
A `301` is a redirect. I would guess that you meant to forward to `https://` and not `http://`

**Reply by notjustcodehq** [reply]
Spot on!!! https:// is what I was missing, thank you. This is really magic!!!
![WebhookAWS](https://user-images.githubusercontent.com/35150537/166912029-d72884e9-8cbc-46df-84b6-8951448134f3.png)
